### PR TITLE
feat: ort v2 BREAKING

### PR DESCRIPTION
**Pull request by joshniemela**
Migrated the code to ort to 2.0.0 based on the breaking API changes
Changed padding strategy and wrote a sequential version of `query_embed`, this gives a speedup of 95%

### CODE REVIEW

**Comment by Anush008** [comment]
Maybe we can dynamic linking mentioned here.
https://ort.pyke.io/setup/linking#load-dynamic-a-glimmer-of-hope

From FastEmbed-go https://github.com/Anush008/fastembed-go/blob/7fe735ed91f0d510ce07b58c6d5dd5ae31fb0218/.github/workflows/release.yml#L30-L34
for `ORT_DYLIB_PATH`.

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ anyhow = { version = "1.0" }
 flate2 = { version = "1.0" }
 minreq = { version = "2.10", default-features = false, features = ["https-rustls"] }
 ndarray = { version = "0.15", default-features = false }
-ort = { version = "1", features = ["load-dynamic"] }
+ort = { version = "2.0.0-alpha.4", features = ["load-dynamic"] }
```
Comment (decahedron1) [inline Cargo.toml:19]: Libraries should only enable features they need.

```toml
[dependencies]
ort = { version = "2.0.0-alpha.4", default-features = false, features = [ "ndarray" ] }

[dev-dependencies]
ort = "2.0.0-alpha.4"
```

**Review by decahedron1 — CHANGES_REQUESTED** [review]

**Comment by decahedron1** [comment]
@Anush008 btw, https://github.com/Anush008/fastembed-rs/pull/12/commits/7e0526a1b8b27c9856ce9e8a1fb521301f9a77d0 has a critical performance fix. @joshniemela and I found that `PaddingStrategy::Fixed` caused `fastembed-rs` to be almost 15x slower than a Python equivalent on shorter sequences. Like the comment says, a new field should be added to `InitOptions` to change the padding strategy if the user wishes (it should also be marked [`#[non_exhaustive]`](https://doc.rust-lang.org/reference/attributes/type_system.html) imo).

**Comment by decahedron1** [comment]
Indeed it significantly improves #9.

`master`
```
init: 138.47ms
passage_embed: 579.84ms
query_embed: 147.26ms
```
`v2`
```
init: 133.73ms
passage_embed: 39.97ms
query_embed: 4.50ms
```

**Comment by joshniemela** [comment]
> Indeed it significantly improves #9.
> 
> `master`
> 
> ```
> init: 138.47ms
> passage_embed: 579.84ms
> query_embed: 147.26ms
> ```
> 
> `v2`
> 
> ```
> init: 133.73ms
> passage_embed: 39.97ms
> query_embed: 4.50ms
> ```

Is it possible to do a test on python as well on the same machine?

**Comment by decahedron1** [comment]
> Is it possible to do a test on python as well on the same machine?

```
init: 118.33ms
passage_embed: 17.55ms
query_embed: 3.51ms
```

**Comment by Anush008** [comment]
Bummer. Should've been a major release.

**Comment by Anush008** [comment]
@joshniemela, @decahedron1 Thanks for the efforts. Super grateful. This is amazing.

**Comment by Anush008** [comment]
After some commit slogging, it is finally up.
https://github.com/Anush008/fastembed-rs/releases/tag/v2.0.0

I've yanked `1.11.0`.

**Comment by prattcmp** [comment]
Thank you for the perf enhancements, but the ONNX runtime upgrade needs to be notated.

This should be released as v2.0.0-alpha or with some other indication that it is not a stable release since it depends on an alpha version of ONNX runtime.

**Comment by decahedron1** [comment]
> This should be released as v2.0.0-alpha or with some other indication that it is not a stable release since it depends on an alpha version of ONNX runtime.

ort v2.0.0-alpha.4 uses ONNX Runtime v1.16.3, so only a minor upgrade as ort 1.16 uses 1.16.0. It does not use an alpha version of ONNX Runtime.

**Comment by prattcmp** [comment]
> > This should be released as v2.0.0-alpha or with some other indication that it is not a stable release since it depends on an alpha version of ONNX runtime.
> 
> ort v2.0.0-alpha.4 uses ONNX Runtime v1.16.3, so only a minor upgrade as ort 1.16 uses 1.16.0. It does not use an alpha version of ONNX Runtime.

Thank you for correcting that the _crate_ is alpha, not the actual runtime. Nevertheless, this release is not stable and should indicate that it is using an alpha version of a crate.

**Pull request merged**